framework/src/org/checkerframework/framework/util/expressionannotations/ExpressionAnnotationHelper.java

```diff
@@ -0,0 +1,715 @@
+    }
+
+    private void standardizeAtm(
+            FlowExpressionContext context,
+            TreePath localScope,
+            AnnotatedTypeMirror type,
+            boolean useLocalScope) {
+        // localScope is null in dataflow when creating synthetic trees for enhanced for loops.
+        if (localScope != null) {
+            new StandardizeTypeAnnotator(context, localScope, useLocalScope).visit(type);
+        }
+    }
+
+    protected String standardizeString(
+            String expression,
+            FlowExpressionContext context,
+            TreePath localScope,
+            boolean useLocalScope) {
+        if (ExpressionAnnotationError.isExpressionError(expression)) {
+            return expression;
+        }
+        try {
+            FlowExpressions.Receiver result =
+                    FlowExpressionParseUtil.parse(expression, context, localScope, useLocalScope);
+            if (result == null) {
+                return new ExpressionAnnotationError(expression, " ").toString();
+            }
+            return result.toString();
+        } catch (FlowExpressionParseUtil.FlowExpressionParseException e) {
+            return new ExpressionAnnotationError(expression, e).toString();
+        }
+    }
+
+    private class StandardizeTypeAnnotator extends AnnotatedTypeScanner<Void, Void> {
+        private final FlowExpressionContext context;
+        private final TreePath localScope;
+        /** Whether or not the expression might contain a variable declared in local scope */
+        private final boolean useLocalScope;
+
+        private StandardizeTypeAnnotator(
+                FlowExpressionContext context, TreePath localScope, boolean useLocalScope) {
+            this.context = context;
+            this.localScope = localScope;
+            this.useLocalScope = useLocalScope;
+        }
+
+        private AnnotationMirror standardizeAnnotation(
+                FlowExpressionContext context,
+                TreePath localScope,
+                AnnotationMirror anno,
+                boolean useLocalScope) {
+            if (!isExpressionAnnotation(anno)) {
+                return null;
+            }
+            List<String> expressionStrings =
+                    AnnotationUtils.getElementValueArray(anno, "value", String.class, true);
+            List<String> vpdStrings = new ArrayList<>();
+            for (String expression : expressionStrings) {
+                vpdStrings.add(standardizeString(expression, context, localScope, useLocalScope));
+            }
+            AnnotationBuilder builder =
+                    new AnnotationBuilder(
+                            factory.getProcessingEnv(), AnnotationUtils.annotationName(anno));
+            builder.setValue("value", vpdStrings);
+            return builder.build();
+        }
+
+        private boolean isExpressionAnnotation(AnnotationMirror anno) {
+            for (Class<? extends Annotation> maybeAnno : expressionAnnos) {
```

Comment:
containsSameByClass takes a List<AnnotationMirror> and a Class<? extends Annotation>.  Here we have a List<Class<?>> and an annotation mirror.